documentation/dataplane/dash-flow-api.md

```diff
@@ -0,0 +1,661 @@
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY0_SIP                | `sai_uint32_t`             | Source IP address in the underlay network                    |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY0_DIP                | `sai_uint32_t`             | Destination IP address in the underlay network               |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY0_DASH_ENCAPSULATION | `sai_dash_encapsulation_t` | Encapsulation method for DASH traffic in the underlay network |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY1_VNI                | `sai_uint32_t`             | Destination VNI in the 2nd underlay network                  |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY1_SIP                | `sai_uint32_t`             | Source IP address in the 2nd underlay network                |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY1_DIP                | `sai_uint32_t`             | Destination IP address in the 2nd underlay network           |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY1_SMAC               | `sai_mac_t`                | Source MAC address in the 2nd underlay network               |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY1_DMAC               | `sai_mac_t`                | Destination MAC address in the 2nd underlay network          |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY1_DASH_ENCAPSULATION | `sai_dash_encapsulation_t` | Encapsulation method for DASH traffic in the 2nd underlay network |
+
+#### Flow overlay rewrite related attributes
+
+These are the related attributes of flow rewrite.
+
+| Attribute name               | Type               | Description                                                  |
+| ---------------------------- | ------------------ | ------------------------------------------------------------ |
+| SAI_FLOW_ENTRY_ATTR_DST_MAC  | `sai_mac_t`        | Destination MAC address for the flow entry.                  |
+| SAI_FLOW_ENTRY_ATTR_SIP      | `sai_ip_address_t` | Source IP address for the flow entry, supporting both IPv4 and IPv6. |
+| SAI_FLOW_ENTRY_ATTR_DIP      | `sai_ip_address_t` | Destination IP address for the flow entry, supporting both IPv4 and IPv6. |
+| SAI_FLOW_ENTRY_ATTR_SIP_MASK | `sai_ip_address_t` | Subnet mask for the source IP address.                       |
+| SAI_FLOW_ENTRY_ATTR_DIP_MASK | `sai_ip_address_t` | Subnet mask for the destination IP address.                  |
+
+### Extra flow metadata
+
+Here are some extra metadata for different purposes.
+
+| Attribute name                      | Type            | Description                                                  |
+| ----------------------------------- | --------------- | ------------------------------------------------------------ |
+| SAI_FLOW_ENTRY_ATTR_VENDOR_METADATA | `sai_u8_list_t` | Vendor-specific metadata that can be attached to the flow entry for custom processing. |
+| SAI_FLOW_ENTRY_ATTR_FLOW_DATA_PB    | `sai_u8_list_t` | The flow data protocol buffer enables high-efficiency creation, retrieval, and communication for a flow entry. |
```

Comment:
I don't think it's a good idea to tie the transport to the SAI struct. Vendor should provide a raw data that can be consumed as is by the other side.

Reply:
i think this attribute "VENDOR_METADAT" didn't break the assumption of "Vendor should provide a raw data that can be consumed as is by the other side.", as this is basically a raw buffer, and we will pass it as it is without the internal knowledge. Do you mind to elaborate this concern a bit more?